root/components/Annotation.js

```diff
@@ -0,0 +1,97 @@
+/*
+ * @flow
+ * Copyright (C) 2018 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {withCatalystContext} from '../context';
+import EditorLink from '../static/scripts/common/components/EditorLink';
+import {l} from '../static/scripts/common/i18n';
+import entityHref from '../static/scripts/common/utility/entityHref';
+import formatUserDate from '../utility/formatUserDate';
+
+import Frag from './Frag';
+
+type Props = {|
+  +$c: CatalystContextT,
+  +annotation: ?AnnotationT,
+  +collapse?: boolean,
+  +entity: CoreEntityT,
+  +numberOfRevisions: number,
+  +showChangeLog?: boolean,
+|};
+
+const Annotation = ({
+  $c,
+  annotation,
+  collapse = false,
+  entity,
+  numberOfRevisions,
+  showChangeLog = false,
+}: Props) => {
+  if (!annotation || !annotation.text || entity.entityType === 'url') {
```

Comment:
Yeah, that was to work around a Flow issue. Since `entity` is a `CoreEntityT`, it expects the component to accept *all* `CoreEntityT` subtypes (including `UrlT`). However, `UrlT` doesn't have a `latest_annotation` property, so it errors below unless this check is added.

The other option is listing out the entity types having annotations explicitly, as I've done in 157e0719d751c3c50ef505a640f307e051ea76a8.